core/dbt/context/common.py

```diff
@@ -211,16 +211,22 @@ def search_package_name(self):
         return self.model.package_name
 
     def add_provider_functions(self, context):
-        context['ref'] = self.provider.ref(
-            self.db_wrapper, self.model, self.config, self.manifest
+        # Generate the builtin functions
+        builtins = dict(
+            ref=self.provider.ref(
+                self.db_wrapper, self.model, self.config, self.manifest),
+            source=self.provider.source(
+                self.db_wrapper, self.model, self.config, self.manifest),
+            config=self.provider.Config(
+                self.model, self.source_config),
+            execute=self.provider.execute
```

Comment:
This is a nit, but can you please use the explicit dict construction syntax like:

```
{
    'ref': self.provider.ref(...),
    'source': ...,
    ...
}
```

instead?

Reply:
Sure. I'm not particularly fussy between the two syntax constructions.